### ISSUE TITLE
Locomotive Scroll cutting of bottom of page

### ISSUE THREAD

**Issue by DennisHallmen** [locomotivemtl/locomotive-scroll#373]
So i am building this page where I want the scrolling to be smooth and where images parallax over each-other but when I implement Locomotive the bottom of the pages just cuts of! And it doesn't matter if i have 12 or 40 images its always just like 1200px from the bottom of the page that cuts off.

Any suggestions?

Here is a preview and live page so you can see what is happening right now.

Preview: https://preview.webflow.com/preview/zenchest?utm_medium=preview_link&utm_source=designer&utm_content=zenchest&preview=57042e62bf640e12cef8bb3f0d966a4a&workflow=preview
Live: https://zenchest.webflow.io/

And if i have to use the ImagesLoaded script which i have read about elsewhere, how do I use it?

Im not a coder:/

**Comment by Dushyant1295** [comment]
#220

**Comment by Dushyant1295** [comment]
You have to update the  locomotive scroll after  all images are loaded,   this thread can help you #220

**Comment by DennisHallmen** [comment]
Yeah i have read that thread  over and over but the thing is that I dont know where exactly to put in the snippet of code or which parts of it to change to my class names. If anyone could just have a check through my webflow project and copy/paste  my code together with the ImagesLoaded code I would be GREATFULL!!!

**Comment by DrKRIOS** [comment]
Locomotive works great, but this bug makes it unusable for me. I'm not a JS developer but I'm highly experienced in css/html. I know my code is not the problem and I guess the issue lays somewhere between loading images and dynamic content scaling.

Can we have any solution that takes this under consideration? An option to "refresh" the loco height or something automatically? That would be great!

**Comment by Le0nidass** [comment]
Hello ! 

For this issues, I use this function ->

Onload of all document ( that prevent footer cut ) 
```
// Scroll update all pages

document.addEventListener('DOMContentLoaded', function() {

    function ScrollUpdateDelay() {
        setTimeout(function(){ scroll.update(); }, 500);
   
    }

    ScrollUpdateDelay();
});
```
And this for dynamic contents ( dropdowns etc.. ), execute with addeventlistener  onclick for exemple ->

```
// Scroll Update trigger

function ScrollUpdateTrig() {
  setTimeout(function(){ scroll.update(); }, 500);

}
```

**Comment by DrKRIOS** [comment]
> Hello !
> 
> For this issues, I use this function ->
> 
> Onload of all document ( that prevent footer cut )
> 
> ```
> // Scroll update all pages
> 
> document.addEventListener('DOMContentLoaded', function() {
> 
>     function ScrollUpdateDelay() {
>         setTimeout(function(){ scroll.update(); }, 500);
>    
>     }
> 
>     ScrollUpdateDelay();
> });
> ```
> 
> And this for dynamic contents ( dropdowns etc.. ), execute with addeventlistener 'click' for exemple ->
> 
> ```
> // Scroll Update trigger
> 
> function ScrollUpdateTrig() {
>   setTimeout(function(){ scroll.update(); }, 500);
> 
> }
> ```

Unfortunately, this only works about 50% of the time... When I refresh my page, it will still occasionally cut half of the content as it used before.

I guess it has something to do with loading multiple images. Yet I think it's would be a killer feature if locomotive covered that with some sort of 'auto refresh' etc.

**Comment by dhanielcodes** [comment]
> Hello !
> 
> For this issues, I use this function ->
> 
> Onload of all document ( that prevent footer cut )
> 
> ```
> // Scroll update all pages
> 
> document.addEventListener('DOMContentLoaded', function() {
> 
>     function ScrollUpdateDelay() {
>         setTimeout(function(){ scroll.update(); }, 500);
>    
>     }
> 
>     ScrollUpdateDelay();
> });
> ```
> 
> And this for dynamic contents ( dropdowns etc.. ), execute with addeventlistener 'click' for exemple ->
> 
> ```
> // Scroll Update trigger
> 
> function ScrollUpdateTrig() {
>   setTimeout(function(){ scroll.update(); }, 500);
> 
> }
> ```

Thanks!, this worked

**Comment by PopCornCommunication** [comment]
I had the same problem but I realized that it was because I was logged on wordpress and the script does not calculate the height of the admin bar

**Comment by dhrutisubham** [comment]
I had my structure of html as
`
<body>
  <div id="preloader"></div>
  <div id="navbar"></div>
  <div id="main" data-scroll-container></div>
  <footer></footer>
</body>
`

Changing it to this structure seems to have solved the problem for me
`
<body>
  <div id="preloader"></div>
  <div id="navbar"></div>
  <div id="main" data-scroll-container>
    
    <footer></footer>
  </div>
</body>
`

**Comment by dhrutisubham** [comment]
I had my structure of html as
```
<body>
  <div id="preloader"></div>
  <div id="navbar"></div>
  <div id="main" data-scroll-container></div>
  <footer></footer>
</body>
```

Changing it to this structure seems to have solved the problem for me
```
<body>
  <div id="preloader"></div>
  <div id="navbar"></div>
  <div id="main" data-scroll-container>
    <!-- 
      Body Content 
    -->
    <footer></footer>
  </div>
</body>
```
